Add wagtail localize syncing

Pomax opened the pull request:
This builds on top of #6111 and adds `wagtail-localize_git` as well as the pontoon syncing script.

This is currently missing the redhook dependency mostly because it seems super sketch unmaintained and pointing to a github commit is a "the owner deleting/renaming their github repo" away from breaking everything, so we need to figure out if we can switch to either something else that managed via a package manager independently of github, or whether we need to do something else.

It's also missing tests, which feels like with the introduction of Redis is pretty important to make sure that even if it's dry run, all the code we're adding gets run, at the very least, even if it doesn't do anything "real", so we can catch any runtime errors that might exist (e.g from bad imports, syntax errors, typos, etc).

Pomax commented:
@TheoChevalier what would be the easiest/best STR for this part of the localization code changes?

Pomax commented on network-api/networkapi/buyersguide/factory.py:
```diff
@@ -37,7 +37,7 @@ def get_random_option(options=[]):
 
 def get_extended_yes_no_value():
-    return get_random_option(['Yes', 'No', 'NA', 'U'])
+    return get_random_option(['Yes', 'No', 'NA', 'CD'])
```
this might be a rebase artifact, and should go away when we rebase `wagtail-localize` on master.

Pomax commented on network-api/networkapi/buyersguide/migrations/0008_auto_20210126_1943.py:
```diff
@@ -0,0 +1,17 @@
+    operations = [
+        migrations.AlterModelOptions(
+            name='update',
+            options={'verbose_name': 'Buyers Guide Product Update', 'verbose_name_plural': 'Buyers Guide Product Updates'},
```
this is the same kind of rebase artifact.

Pomax commented on network-api/networkapi/wagtailpages/pagemodels/products.py:
```diff
@@ -368,6 +368,11 @@ class ProductPage(AirtableMixin, FoundationMetadataPageMixin, Page):
         related_name='votes',
     )
 
+    override_translatable_fields = [
+        SynchronizedField('votes'),
+        SynchronizedField('creepiness_value'),
+    ]
+
```
I remember you saying we no longer need this, @TheoChevalier?

TheoChevalier replied:
correct

Pomax replied:
```suggestion
```

TheoChevalier commented:
> @TheoChevalier what would be the easiest/best STR for this part of the localization code changes?

Once the Redis issue/concern has been addressed, I guess:

- Setup a dummy repo to `WAGTAILLOCALIZE_GIT_URL`
- Add a new locale to Settings > Locales (e.g. French), syncing from English
- Translate a page in French
- Check that content is pushed to the GitHub repo
- Clone the repo locally, and edit the .po file with [PoEdit](https://poedit.net/download), to make a dummy translation (may need to define French as target language)
- Commit the changes
- Wait 10 minutes (not sure if the cronjob is set automatically), alternatively open the Settings > Pontoon and force sync
- Check the French translations appear on the French page

KalobTaulien commented:
There's a migration that needs to be merged when I run `inv new-env`. 

```
CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0031_auto_20210122_0006, 0032_productpage_user_friendly_privacy_policy_helptext in wagtailpages; 0008_auto_20210127_0010, 0008_auto_20210126_1943 in buyersguide).
To fix them run 'python manage.py makemigrations --merge'
```

KalobTaulien reviewed (dismissed):
[no longer applicable]

Pomax commented:
@TheoChevalier so I've created a dummy repo over on https://github.com/Pomax/test-repo-for-wagtail-localization and updated my `.env` to say `WAGTAILLOCALIZE_GIT_URL=git@github.com:Pomax/test-repo-for-wagtail-localization.git`, then I:

- ran `inv new-env`
- logged into the admin and added the German locale (because that required less clicking)
- loaded the "homepage" -> "what we do" page
- clicked "translate this page" and picked "German"

This creates a german page copy for localizing, but I don't see any console notices about Wagtail trying to upload anything to the git repo: which additional steps are necessary to trigger a git upload? (Also, where would one set permissions to make that git push be accepted? Because I assume you can't just push to a repo without authenticating)

TheoChevalier commented:
> @TheoChevalier so I've created a dummy repo over on https://github.com/Pomax/test-repo-for-wagtail-localization and updated my `.env` to say `WAGTAILLOCALIZE_GIT_URL=git@github.com:Pomax/test-repo-for-wagtail-localization.git`, then I:
> 
>     * ran `inv new-env`
> 
>     * logged into the admin and added the German locale (because that required less clicking)
> 
>     * loaded the "homepage" -> "what we do" page
> 
>     * clicked "translate this page" and picked "German"
> 
> 
> This creates a german page copy for localizing, but I don't see any console notices about Wagtail trying to upload anything to the git repo: which additional steps are necessary to trigger a git upload? (Also, where would one set permissions to make that git push be accepted? Because I assume you can't just push to a repo without authenticating)

Nice! I think you just need to set a [deploy key](https://docs.github.com/en/developers/overview/managing-deploy-keys#deploy-keys) with read/write permissions on the repo, I can see it’s what is being used to commit to the donate repos. Lucie set these up, so I’m not entirely sure how it’s set up on the server side, but likely just loading the private key is enough?

Pomax commented:
Closing because https://github.com/mozilla/foundation.mozilla.org/issues/6711 is the new hot
